app/src/main/java/com/example/filemanager/ListAdapter.kt

```diff
@@ -9,18 +9,19 @@ import android.widget.TextView
 import androidx.core.content.ContextCompat
 import androidx.recyclerview.widget.RecyclerView
 
-internal class ListAdapter(
+class ListAdapter(
     private val context: Context,
-    private val list: MutableList<ListModel>,
     private val itemClickListener: ItemClickListener
-) :
-    RecyclerView.Adapter<ListAdapter.ViewHolder>() {
+) : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
 
-    internal inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+    private val list = mutableListOf<ListModel>()
```

Comment:
Can use a better name for this `val`, from it's name should be clear what data it holds in it, `files`, `filesList`.